src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2618,27 +2624,40 @@ static void WindowRideConstructionUpdateWidgets(rct_window* w)
     if (IsTrackEnabled(TRACK_SLOPE))
     {
         window_ride_construction_widgets[WIDX_SLOPE_DOWN].type = WindowWidgetType::FlatBtn;
-        window_ride_construction_widgets[WIDX_SLOPE_UP].type = WindowWidgetType::FlatBtn;
-    }
-    if (IsTrackEnabled(TRACK_HELIX_SMALL) && _currentTrackBankEnd != TRACK_BANK_NONE
-        && _currentTrackSlopeEnd == TRACK_SLOPE_NONE)
-    {
-        if (_currentTrackCurve >= TRACK_CURVE_LEFT && _currentTrackCurve <= TRACK_CURVE_RIGHT_SMALL)
+
+        if (!ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_UP_INCLINE_REQUIRES_LIFT)
+            || ((_previousTrackSlopeEnd == TRACK_SLOPE_NONE && _previousTrackBankEnd == TRACK_BANK_NONE)
+                || (_currentTrackLiftHill & CONSTRUCTION_LIFT_HILL_SELECTED))
+            || gCheatsEnableAllDrawableTrackPieces)
         {
-            // Enable helix
-            window_ride_construction_widgets[WIDX_SLOPE_DOWN_STEEP].type = WindowWidgetType::FlatBtn;
-            if (rideType != RIDE_TYPE_SPLASH_BOATS && rideType != RIDE_TYPE_RIVER_RAFTS)
-                window_ride_construction_widgets[WIDX_SLOPE_UP_STEEP].type = WindowWidgetType::FlatBtn;
+            window_ride_construction_widgets[WIDX_SLOPE_UP].type = WindowWidgetType::FlatBtn;
         }
     }
 
-    if (IsTrackEnabled(TRACK_SLOPE_STEEP))
+    if (IsTrackEnabled(TRACK_HELIX_SMALL) && _currentTrackBankEnd != TRACK_BANK_NONE
+        && _currentTrackSlopeEnd == TRACK_SLOPE_NONE
+        && (_currentTrackCurve >= TRACK_CURVE_LEFT && _currentTrackCurve <= TRACK_CURVE_RIGHT_SMALL))
+    {
+        // Enable helix
+        window_ride_construction_widgets[WIDX_SLOPE_DOWN_STEEP].type = WindowWidgetType::FlatBtn;
+        if (gCheatsEnableAllDrawableTrackPieces
+            || !ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_UP_INCLINE_REQUIRES_LIFT))
+            window_ride_construction_widgets[WIDX_SLOPE_UP_STEEP].type = WindowWidgetType::FlatBtn;
+    }
+    else if (IsTrackEnabled(TRACK_SLOPE_STEEP))
     {
         window_ride_construction_widgets[WIDX_SLOPE_DOWN_STEEP].type = WindowWidgetType::FlatBtn;
-        if (rideType != RIDE_TYPE_SPLASH_BOATS && rideType != RIDE_TYPE_RIVER_RAFTS)
+        if (!ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_GENTLE_UP_INCLINE_ONLY)
+            || gCheatsEnableAllDrawableTrackPieces)
```

Comment:
This will show empty sprites for the Splash Boats if steep up is selected.

Rather than introducing a ride type flag, we should perhaps split the TRACK_SLOPE_STEEP flag into ones for steep down and steep up. That would allow us to do this more cleanly.

Reply:
> Rather than introducing a ride type flag, we should perhaps split the TRACK_SLOPE_STEEP flag into ones for steep down and steep up. That would allow us to do this more cleanly.

Sounds like a good idea, but is it enough if I just remove all changes to the splash boats and leave just the RIDE_FLAG_INCLINE_REQUIRES_LIFT flag?

Reply:
Yeah, just limit this PR to `RIDE_TYPE_FLAG_UP_INCLINE_REQUIRES_LIFT`, the rest should be taken care of in https://github.com/OpenRCT2/OpenRCT2/pull/16836.